core/arch/arm/include/kernel/generic_boot.h

```diff
@@ -31,10 +31,12 @@
 
 #if defined(CFG_WITH_ARM_TRUSTED_FW)
 uint32_t cpu_on_handler(uint32_t a0, uint32_t a1);
-uint32_t *generic_boot_init_primary(uint32_t pageable_part);
+uint32_t *generic_boot_init_primary(uint32_t pageable_part, uint32_t unused,
+				    uint32_t fdt);
```

Comment:
`paddr_t fdt`? Otherwise you still can't get full 64-bit addresses from the firmware (what if the DT blob is at a physical address > 4GB)...